integration_tests/framework/xml_test_runner.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Test main and flags."""
+
+import logging
+import os
+import sys
+import time
+import traceback
+import unittest
+
+from xml.etree import ElementTree
+
+XML_OUTPUT_FILE = 'XML_OUTPUT_FILE'
+DEFAULT_SUBTEST = 'Test Case'
+
+
+def _GetMillis():
+  return int(time.time() * 1000)
+
+
+def _Typename(cls):
+  return '%s.%s' % (cls.__module__, cls.__name__)
+
+
+def _AddProperties(element, props):
+  if props:
+    sub = ElementTree.SubElement(element, 'properties')
+    for k, v in props.items():
+      ElementTree.SubElement(sub, 'property', name=k, value=v)
+
+
+class XmlTestRunner(object):
+  """Runner producing test xml output."""
```

Comment:
Can you describe the target consumer of this XML file? Which schema does it use?

Reply:
I copied this from the old code base. I'm not sure whether we really need it. Let's see if we can get rid of it. The less, the merrier. I think it's only needed for Sponge compatibility.

Turns out the XML stuff wasn't really needed. I've reduced it to the minimum, i.e. the stuff that makes the logs nicer and prints errors.